pertpy/tools/_differential_gene_expression/_base.py

```diff
@@ -449,20 +505,449 @@ def _map_genes_categories_highlight(
+        self,
+        results_df: pd.DataFrame,
+        *,
+        var_names: Sequence[str] = None,
+        n_top_vars: int = 15,
+        y_label: str = "Log2 fold change",
+        figsize: tuple[int, int] = (10, 5),
+        show: bool = True,
+        return_fig: bool = False,
+        **barplot_kwargs,
+    ) -> Figure | None:
+        """Plot a metric from the results as a bar chart, optionally with additional information about paired samples in a scatter plot.
+
+        Args:
+            results_df: DataFrame with results from DE analysis.
+            pairedby: Column in results_df containing information about paired samples.
+            var_names: Variables to plot. If None, the top n_top_vars variables based on the log2 fold change are plotted.
+            n_top_vars: Number of top variables to plot. The top and bottom n_top_vars variables are plotted, respectively.
+            y_label: Label for the y-axis.
+            figsize: Size of the figure.
+            {common_plot_args}
+            **barplot_kwargs: Additional arguments for seaborn.barplot.
+
+        Returns:
+            If `return_fig` is `True`, returns the figure, otherwise `None`.
+
+        Examples:
+            >>> # Example with EdgeR
+            >>> import pertpy as pt
+            >>> adata = pt.dt.zhang_2021()
+            >>> adata.layers["counts"] = adata.X.copy()
+            >>> ps = pt.tl.PseudobulkSpace()
+            >>> pdata = ps.compute(
+            ...     adata,
+            ...     target_col="Patient",
+            ...     groups_col="Cluster",
+            ...     layer_key="counts",
+            ...     mode="sum",
+            ...     min_cells=10,
+            ...     min_counts=1000,
+            ... )
+            >>> edgr = pt.tl.EdgeR(pdata, design="~Efficacy+Treatment")
+            >>> edgr.fit()
+            >>> res_df = edgr.test_contrasts(
+            ...     edgr.contrast(column="Treatment", baseline="Chemo", group_to_compare="Anti-PD-L1+Chemo")
+            ... )
+            >>> edgr.plot_fold_change(res_df)
+
+        Preview:
+            .. image:: /_static/docstring_previews/de_fold_change.png
+        """
+        if var_names is None:
+            var_names = results_df.sort_values("log_fc", ascending=False).head(n_top_vars)["variable"].tolist()
+            var_names += results_df.sort_values("log_fc", ascending=True).head(n_top_vars)["variable"].tolist()
+            assert len(var_names) == 2 * n_top_vars
+
+        df = results_df[results_df["variable"].isin(var_names)]
+        df.sort_values("log_fc", ascending=False, inplace=True)
+
+        max_fc = max(df["log_fc"])
+        min_fc = min(df["log_fc"])
+
+        def value_to_color(val):
+            if val > 0:
+                return plt.cm.Reds(val / max_fc)
+            elif val < 0:
+                return plt.cm.Blues(val / min_fc)
+            else:
+                return "grey"
+
+        df["color"] = df["log_fc"].apply(value_to_color)
```

Comment:
This certainly works, but why manually encode the colors instead of just specifying a diverging colormap to seaborn?

Reply:
Yes, true. I've changed it!